recipes/onnxruntime/all/conanfile.py

```diff
@@ -41,10 +41,18 @@ def _min_cppstd(self):
 
     @property
     def _compilers_minimum_version(self):
+        if Version(self.version) < "1.16.0":
+            return {
+                "Visual Studio": "16",
+                "msvc": "192",
+                "gcc": "7",
+                "clang": "5",
+                "apple-clang": "10",
+            }
         return {
-            "Visual Studio": "16",
-            "msvc": "192",
-            "gcc": "7",
+            "Visual Studio": "17",
+            "msvc": "193",
+            "gcc": "9",
```

Comment:
Does onnxruntime declare what minimum versions are supported? Or the gcc7 build failed?

Reply:
GCC is the only one checked in their CMakeList.txt, they bumped the check from 7 to 9.
For MSVC, it's not enforced but they dropped support for 2019 in their CI build driver & went to 2022.
At least one issue in the GitHub confirms that support for VS 2019 is dropped: https://github.com/microsoft/onnxruntime/issues/17693#issuecomment-1734645957

Reply:
We are moving the C++ standard from 17 to 20.  C++20 needs VS 2023 or later.